ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/composables/Footer.kt

```diff
@@ -83,18 +83,31 @@ private fun Footer(
 ) {
     val context = LocalContext.current
     val uriHandler = LocalUriHandler.current
-
+    val shouldDisplayAllPlansButton = mode == PaywallMode.FOOTER_CONDENSED && allPlansTapped != null
+    val anyButtonDisplayed = shouldDisplayAllPlansButton ||
+        configuration.displayRestorePurchases ||
+        configuration.termsOfServiceURL != null ||
+        configuration.privacyURL != null
+    val bottomMargin = if (anyButtonDisplayed) {
+        0.dp
+    } else {
+        UIConstant.defaultVerticalSpacing * 2
+    }
     Row(
         modifier = Modifier
             .fillMaxWidth()
             .height(intrinsicSize = IntrinsicSize.Min)
-            .padding(horizontal = UIConstant.defaultHorizontalPadding),
+            .padding(
+                start = UIConstant.defaultHorizontalPadding,
+                end = UIConstant.defaultHorizontalPadding,
+                bottom = bottomMargin,
+            ),
         horizontalArrangement = Arrangement.Center,
         verticalAlignment = Alignment.CenterVertically,
     ) {
         val color = colors.text1
 
-        if (mode == PaywallMode.FOOTER_CONDENSED && allPlansTapped != null) {
+        if (shouldDisplayAllPlansButton && allPlansTapped != null) {
```

Comment:
This is a bit stupid but need to add the `allPlansTapped != null` check to make sure the compiler picks up it's not null within the if. We don't have Kotlin 2 😢